internal/replacements/replacements_test.go

```diff
@@ -296,6 +296,23 @@ var _ = Describe("replace_misconfigured_pods", func() {
 			})
 		})
 
+		Context("when the securityContext doesn't match and the PodSpecHash doesn't match", func() {
```

Comment:
this is wrong as the setting is for all changes for this test case, I will add another which does not have this set to true (default is false)